pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -1829,6 +1829,7 @@ class Timedelta(_Timedelta):
                 return NaT
 
             return _timedelta_from_value_and_reso(
+                Timedelta,
```

Comment:
should this be `type(self)`?

Reply:
I had added a question re: this but somehow github ate it, weird..

Anyway, here is my concern, if I make the change you are suggesting

```
td1 = MyTimedeltaSubclass("10 minutes")
td2 = Timedelta ("10 minutes")

then result types will be inconsistent
td1 + td2 -> MyTimedeltaSubclass
td2 + td1 -> Timedelta
```

Maybe that is fine, but it also kinda weirded me out, which is why I wanted to ask you. It feels a little bit like it should be the responsibility of the subclass to implement the custom __add__ behaviour. 

Wondering: does the rest of pandas respect subclasses in this way?

Reply:
Should I go ahead and create an issue for this?